crates/cli/src/commands/apply_pattern.rs

```diff
@@ -159,7 +159,7 @@ pub struct ApplyPatternArgs {
     #[clap(long)]
     ai: bool,
     /// Change the default language to use for the pattern (if unset, JavaScript is used by default)
-    #[clap(long = "language", alias = "lang")]
+    #[clap(long = "language", alias = "lang", value_parser = PatternLanguage::value_parser())]
```

Comment:
Note that we want to preserve the skipped variants: https://github.com/getgrit/gritql/blob/5e7ca26133fd2d847f73966ca5c45b121b132faf/crates/language/src/target_language.rs#L52

Please add a test to make sure these variants aren't included as options.

Reply:
I had added the opposite test by the sounds of it - I had added a test to verify that `javascript` was a valid alias:

https://github.com/getgrit/gritql/blob/a5887a7d44bdc884c94ef076e7edc5d75ad1788d/crates/cli_bin/tests/apply.rs#L2856-L2869

Reply:
For now I assumed all extensions and all lowercase names of the enum variants should be "by default" become allowed language aliases e.g. `javascript` and `phponly` are valid aliases (of `js` and `php` respectively), but perhaps at least `phponly` is one you wouldn't want to include as it is skipped and it sounds a bit like an internal name rather than a user facing name?

perhaps I remove this `to_str` method and make the aliases more explicit, by adding `javascript` explicitly to `LANGUAGE_ALIASES` (assuming that `javascript` is a desired alias for `js`) but don't add all the lowercase names from the enum variants and honour the skipped variants by using `fmt` instead of `to_str` inside of `possible_values()` to generate the list of possible values, while honouring skipped variants and renamed variants etc.

Reply:
Sorry, I should have been clearer:

* `javascript` *should* be a valid alias of `js` so it will work as a CLI option.
* I don't want to *list* all aliases/flavors in the "help" - it makes the help string a lot more complicated / hard to follow. Instead, the help (which I think just comes from possible_values) should only list the primary name for each language.

> so this command should fail? It currently succeeds with my PR changes (from my comment):
No, it should pass.

Reply:
test added to ensure that `markdowninline` is an invalid alias as it is a skipped variant: 567a8a0